src/core/public/chrome/nav_links/nav_link.ts

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * @public
+ */
+export interface ChromeNavLink {
+  /**
+   * A unique identifier for looking up links.
+   */
+  readonly id: string;
+
+  /**
+   * Indicates whether or not this app is currently on the screen.
+   *
+   * NOTE: remove this when ApplicationService is implemented and managing apps.
+   */
+  readonly active?: boolean;
+
+  /**
+   * Disables a link from being clickable.
+   *
+   * NOTE: this is only used by the ML and Graph plugins currently. They use this field
```

Comment:
This comment is not technically true from the point of view of ML.
If the user's platinum license has expired, ML will still be available in the nav bar but some features will be disabled within the ML app, such as creating new jobs.

The disabling of the ML icon in the nav bar originally happened if we could not load the x-pack license information. This appears to be broken, and has been broken before this PR.
We originally copied this behaviour from another x-pack app, Graph i believe, but it was a long time ago.